node_modules/gh-users/lib/api.js

```diff
@@ -324,6 +335,243 @@ var updateAdminStatus = module.exports.updateAdminStatus = function(ctx, userId,
+ *
+ * When importing a set of users using the local authentication strategy, the CSV format should be the following:
+ *
+ *  `displayName, email, password`
+ *
+ * @param  {Context}        ctx                             Standard context containing the current user and the current app
+ * @param  {File}           file                            The CSV file to import
+ * @param  {Number}         appId                           The id of the application into which the users should be imported
+ * @param  {String}         authenticationStrategy          The authentication strategy for the user. One of `local` or `shibboleth`
+ * @param  {Boolean}        [forceProfileUpdate]            Whether the user information should be updated, even when other user information is already present. Defaults to `false`
+ * @param  {Function}       callback                        Standard callback function
+ * @param  {Object}         callback.err                    An error that occurred, if any
+ */
+var importUsers = module.exports.importUsers = function(ctx, file, appId, authenticationStrategy, forceProfileUpdate, callback) {
+    forceProfileUpdate = forceProfileUpdate || false;
+    forceProfileUpdate = GrasshopperUtil.getBooleanParam(forceProfileUpdate);
+    appId = GrasshopperUtil.getNumberParam(appId);
+
+    var validator = new Validator();
+    validator.check(null, {'code': 401, 'msg': 'Only authenticated users can import users'}).isLoggedInUser(ctx);
+    validator.check(authenticationStrategy, {'code': 400, 'msg': 'A valid authentication strategy must be provided'}).isIn(['local', 'shibboleth']);
+    validator.check(null, {'code': 400, 'msg': 'A valid CSV file must be provided'}).isObject(file);
+    if (file) {
+        validator.check(file.file, {'code': 400, 'msg': 'Missing path to the CSV file'}).notEmpty();
+    }
+    validator.check(appId, {'code': 400, 'msg': 'A valid application id must be provided'}).isInt();
+    validator.check(null, {'code': 400, 'msg': 'A valid profile update flag must be provided'}).isBoolean(forceProfileUpdate);
+    if (validator.hasErrors()) {
+        _cleanUpCSVFile(file, function() {
+            return callback(validator.getFirstError());
+        });
+        return;
+    }
+
+    // Ensure the application exists
+    AppsAPI.getApp(ctx, appId, function(err, app) {
+        if (err) {
+            return callback(err);
+        }
+
+        // Ensure the current user can import users
+        UsersAuthz.canImportUsers(ctx, app, function(err, canImportUsers) {
+            if (err) {
+                return callback(err);
+            } else if (!canImportUsers) {
+                log().warn({'app': appId, 'actor': ctx.user.id}, 'Unauthorized attempt at importing users');
+                return callback({'code': 401, 'msg': 'You are not allowed to import users'});
+            }
+
+            // Construct the CSV parser
+            var parser = csv.parse({'trim': true}, function(err, users) {
```

Comment:
Can we be more clear that these are `userRows`, and only switch to using pure `user` terminology if we're dealing with a `user` model object.